src/GoTrueClient.ts

```diff
@@ -195,7 +208,57 @@ export default class GoTrueClient {
           scopes: options.scopes,
         })
       }
-      throw new Error(`You must provide either an email or a third-party provider.`)
+      throw new Error(`You must provide either an email, phone number or a third-party provider.`)
+    } catch (error) {
+      return { data: null, user: null, session: null, error }
+    }
+  }
+
+  /**
+   * Log in a user given a User supplied OTP received via mobile.
+   * @param phone The user's phone number.
+   * @param token The user's password.
+   * @param redirectTo A URL or mobile address to send the user to after they are confirmed.
+   */
+  async verifyOTP(
```

Comment:
left this one generic so we have the option to add other types of OTP tokens later